enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,329 @@
+  In order to prevent this, WICD can set the Windows service description to "openshift-managed", so that it
+  can search for services and remove them. In order to facilitate upgrades from a version without WICD, this string
+  should be added to all existing services in 4.10, one major version ahead of WICD's target release.
+
+## Design Details
+
+### Services ConfigMap
+
+WMCO will create and maintain an immutable ConfigMap with the naming scheme:
+`services-<MajorVersion>-<MinorVersion>-<PatchVersion>-<Commit>`. For example, if WMCO was built from commit a7b5
+of WMCO version 5.0.0, the created ConfigMap would be named `services-5-0-0-a7b5`.
+
+The purpose of this ConfigMap is to provide WICD with the specifications required to configure a Node. The ConfigMap
+will be watched via the existing ConfigMap controller. WMCO is the owner of the ConfigMap and the source of the data
+present within it. If an entity attempts to create a services ConfigMap with incorrect values, WMCO will delete it.
+If it is deleted, WMCO will re-create it.
+
+The services ConfigMap contains two keys: `services`, and `files`. The values for both keys are JSON objects.
+The `services` key contains all data required to configure the required Windows services on any instance that is to be
+added to the cluster as a Node. The proposed schema is as follows:
+```json
+[
+  {
+    "name": "name of the Windows service",
+    "command": "command that will be executed. This could potentially include strings whose value will be derived from nodeVariablesinCommand and powershellVariablesinCommand.",
+    "dependencies": [
+      "name of a service that this service is dependent on"
+    ],
+    "nodeVariablesinCommand": {
+      "name": "string within command field that will be substituted",
+      "jsonPathNodeObject": "The jsonPath of a field within the instance's Node object"
+    },
+    "powershellVariablesinCommand": {
+      "name": "string within command field that will be substituted",
+      "path": "location of the PowerShell script to be run, in order to get its output"
+    },
+    "priority": "integer that will be used to order the creation of the services, priority 1 is created first"
+  }
+]
+```
+
+The nodeVariablesToSubsitute and powershellVariablesinCommand fields will be used by WICD when processing the
+ConfigMap. They are needed as some Windows services have arguments that require values sourced from either the
+Node object associated with the instance, or data from the instance itself. These fields allow for a ConfigMap that is
+generic enough to apply to all Windows instances being configured, by providing WICD instructions for how to retrieve
+certain values. For these fields, a retry period will be necessary for when the information is not available yet. For
+example, when waiting for an annotation to be applied to the Node object. When making use of a PowerShell script,
+retries should be present within the script, if necessary.
```

Comment:
What about with `nodeVariablesCommand`? Won't we have to wait for those too?

Reply:
I call that out: `For example, when waiting for an annotation to be applied to the Node object`